src/bucket/BucketList.cpp

```diff
@@ -916,6 +919,26 @@ BucketList::scanForEviction(Application& app, AbstractLedgerTxn& ltx,
                 if (evictionIter.bucketListLevel == kNumLevels)
                 {
                     evictionIter.bucketListLevel = firstScanLevel;
+
+                    // If eviction metrics are not null, we have accounted for a
+                    // complete cycle and should log the metrics
+                    if (mEvictionMetrics)
+                    {
+                        evictionCyclePeriodCounter.set_count(
+                            ledgerSeq -
+                            mEvictionMetrics->evictionCycleStartLedger);
+
+                        auto averageAge =
+                            mEvictionMetrics->numEntriesEvicted == 0
+                                ? 0
+                                : mEvictionMetrics->evictedEntriesAgeSum /
+                                      mEvictionMetrics->numEntriesEvicted;
+                        averageEvictedEntryAgeCounter.set_count(averageAge);
+                    }
+
+                    // Reset metrics at beginning of new eviction cycle
+                    mEvictionMetrics = std::make_optional<EvictionMetrics>();
+                    mEvictionMetrics->evictionCycleStartLedger = ledgerSeq;
```

Comment:
this won't record the first eviction scan; we could initialize evictionCycleStartLedger at the beginning of eviction scan to fix that, right?

Reply:
We don't actually have a code path that checks if we're at the start of an eviction scan. Personally, I prefer it as is. I like that we record and re-initialize our metrics at the same point, and would rather not add a branch that will only every be called once when we upgrade to 20. Also, the first eviction cycle metrics will be pretty meaningless anyway since there will not yet be any entries eligible for eviction.

Reply:
I don't feel strongly, but I thought it's possible to evict entries in the first cycle if some buckets spill faster and cause eviction scan to get reset. I'm fine with keeping the current approximation though.